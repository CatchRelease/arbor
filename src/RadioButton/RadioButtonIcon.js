import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';

const iconFontSize = '16px';

const RadioButtonIcon = ({ checked }) => {
  if (checked) {
    return <Icon name="radio-checked" fontSize={iconFontSize} />;
  }

  return null;
};

RadioButtonIcon.propTypes = {
  /**
   * Whether or not the checkbox is checked.
   */
  checked: PropTypes.bool
};

RadioButtonIcon.defaultProps = {
  checked: false
};

export default RadioButtonIcon;
