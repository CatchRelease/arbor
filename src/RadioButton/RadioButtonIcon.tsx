import { VFC } from 'react';

import Icon from '../Icon';

const iconFontSize = '16px';

type Props = {
  checked?: boolean;
};

const RadioButtonIcon: VFC<Props> = ({ checked }) => {
  if (checked) {
    return <Icon name="radio-checked" fontSize={iconFontSize} />;
  }

  return null;
};

export default RadioButtonIcon;
