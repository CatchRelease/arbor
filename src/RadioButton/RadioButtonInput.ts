import styled from '@emotion/styled';

import Box from '../Box';

const RadioButtonInput = styled(Box)`
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

RadioButtonInput.defaultProps = {
  as: 'input'
};

export default RadioButtonInput;
