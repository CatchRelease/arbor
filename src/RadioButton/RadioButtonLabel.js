import styled from '@emotion/styled';

import Flex from '../Flex';

const RadioButtonLabel = styled(Flex)`
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  line-height: ${({ theme }) => theme.lineHeights.small};
`;

RadioButtonLabel.defaultProps = {
  as: 'label'
};

export default RadioButtonLabel;
