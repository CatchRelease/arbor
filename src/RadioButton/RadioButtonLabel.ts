import styled from '@emotion/styled';

import Flex, { FlexProps } from '../Flex';

export type Props = FlexProps;

const RadioButtonLabel = styled(Flex)`
  cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
  line-height: ${({ theme }) => theme.lineHeights.small};
`;

RadioButtonLabel.defaultProps = {
  as: 'label'
};

export default RadioButtonLabel;
