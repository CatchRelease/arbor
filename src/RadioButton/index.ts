export { default, Props as RadioButtonProps } from './RadioButton';
