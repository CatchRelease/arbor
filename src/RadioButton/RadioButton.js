import PropTypes from 'prop-types';

import Icon from '../Icon';
import Text from '../Text';

import RadioButtonIcon from './RadioButtonIcon';
import RadioButtonInput from './RadioButtonInput';
import RadioButtonLabel from './RadioButtonLabel';
import StyledRadioButton from './StyledRadioButton';

const RadioButton = ({
  id,
  name,
  label,
  disabled,
  isInvalid,
  checked,
  onChange,
  value,
  icon,
  ...props
}) => (
  <RadioButtonLabel alignItems="center" my="regular" {...props}>
    <RadioButtonInput
      id={id}
      type="radio"
      name={name}
      value={value}
      checked={checked}
      onChange={onChange}
      disabled={disabled}
      aria-invalid={isInvalid}
    />

    <StyledRadioButton {...{ disabled, checked }}>
      <RadioButtonIcon {...{ checked }} />
    </StyledRadioButton>

    {icon && (
      <Icon
        fontSize="24px"
        name={icon}
        color={disabled ? 'icon.disabled' : 'icon.default'}
        ml="smaller"
      />
    )}

    <Text
      as="span"
      fontSize="size4"
      color={disabled ? 'text.disabled' : 'text.default'}
      ml={icon ? 'smallest' : 'smaller'}
    >
      {label}
    </Text>
  </RadioButtonLabel>
);

RadioButton.propTypes = {
  /**
   * The checked attribute of the checkbox.
   */
  checked: PropTypes.bool,

  /**
   * When true, the checkbox is disabled.
   */
  disabled: PropTypes.bool,

  /**
   * The icon to display alongside the label.
   */
  icon: PropTypes.elementType,

  /**
   * The id attribute of the checkbox.
   */
  id: PropTypes.string.isRequired,

  /**
   * When true, the aria-invalid attribute is true.
   * Used for accessibility.
   */
  isInvalid: PropTypes.bool,

  /**
   * Label of the checkbox.
   */
  label: PropTypes.node.isRequired,

  /**
   * The name attribute of the checkbox.
   */
  name: PropTypes.string.isRequired,

  /**
   * Function called when state changes.
   */
  onChange: PropTypes.func,

  /**
   * The value attribute of the checkbox.
   */
  value: PropTypes.string
};

RadioButton.defaultProps = {
  checked: false,
  disabled: false,
  icon: undefined,
  isInvalid: false,
  onChange: () => {},
  value: ''
};

export default RadioButton;
