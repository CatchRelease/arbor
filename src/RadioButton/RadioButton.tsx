import { FormEventHandler, ReactNode, VFC } from 'react';

import Icon from '../Icon';
import Text from '../Text';

import RadioButtonIcon from './RadioButtonIcon';
import RadioButtonInput from './RadioButtonInput';
import RadioButtonLabel from './RadioButtonLabel';
import StyledRadioButton from './StyledRadioButton';

type Props = {
  checked?: boolean;
  disabled?: boolean;
  icon?: string;
  id: string;
  isInvalid?: boolean;
  label: ReactNode;
  name: string;
  onChange?: FormEventHandler<HTMLElement>;
  value?: string;
};

const RadioButton: VFC<Props> = ({
  checked = false,
  disabled = false,
  icon,
  id,
  isInvalid = false,
  label,
  name,
  onChange = () => {},
  value = '',
  ...props
}) => (
  <RadioButtonLabel alignItems="center" my="regular" {...props}>
    <RadioButtonInput
      id={id}
      type="radio"
      name={name}
      value={value}
      checked={checked}
      onChange={onChange}
      disabled={disabled}
      aria-invalid={isInvalid}
    />

    <StyledRadioButton {...{ disabled, checked }}>
      <RadioButtonIcon {...{ checked }} />
    </StyledRadioButton>

    {icon && (
      <Icon
        fontSize="24px"
        name={icon}
        color={disabled ? 'icon.disabled' : 'icon.default'}
        ml="smaller"
      />
    )}

    <Text
      as="span"
      fontSize="size4"
      color={disabled ? 'text.disabled' : 'text.default'}
      ml={icon ? 'smallest' : 'smaller'}
    >
      {label}
    </Text>
  </RadioButtonLabel>
);

export default RadioButton;
