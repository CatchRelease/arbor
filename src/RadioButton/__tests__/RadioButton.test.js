import React from 'react';

import createWithTheme from '../../../utils/createWithTheme';
import RadioButton from '../RadioButton';

describe('<RadioButton />', () => {
  const baseProps = {
    id: 'example',
    name: 'example',
    label: 'My RadioButton',
    value: '1'
  };

  const createWithProps = (additionalProps) => {
    const props = { ...baseProps, ...additionalProps };

    return createWithTheme(<RadioButton {...{ ...props }} />);
  };

  it('properly renders a radio button', () => {
    const tree = createWithProps();

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a checked radio button', () => {
    const tree = createWithProps({ checked: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a disabled checkbox', () => {
    const tree = createWithProps({ disabled: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a disabled checked radio button', () => {
    const tree = createWithProps({ disabled: true, checked: true });

    expect(tree).toMatchSnapshot();
  });

  it('properly renders a radio button with an icon', () => {
    const tree = createWithProps({ icon: 'cr-logo' });

    expect(tree).toMatchSnapshot();
  });
});
