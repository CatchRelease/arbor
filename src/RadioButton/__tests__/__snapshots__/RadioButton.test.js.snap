// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> properly renders a checked radio button 1`] = `
.emotion-8 {
  font-family: Motiva Sans;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  line-height: 1.25;
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #FFFFFF;
  border: 1px solid #989A9C;
  border-radius: 50%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  color: #081D17;
}

.emotion-2 {
  font-size: 16px;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={false}
  >
    <i
      className="ar ar-radio-checked emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My RadioButton
  </span>
</label>
`;

exports[`<RadioButton /> properly renders a disabled checkbox 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #FFFFFF;
  border: 1px solid #989A9C;
  border-radius: 50%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  border-color: #E0E1E2;
  color: #FFFFFF;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: rgba(120,121,125,0.7);
  font-size: 0.875rem;
  line-height: 1.25;
  margin-left: 0.5rem;
}

<label
  className="emotion-6 emotion-7"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={true}
    id="example"
    name="example"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={true}
  />
  <span
    className="emotion-4 emotion-5"
    color="text.disabled"
    fontSize="size4"
  >
    My RadioButton
  </span>
</label>
`;

exports[`<RadioButton /> properly renders a disabled checked radio button 1`] = `
.emotion-8 {
  font-family: Motiva Sans;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  font-size: 16px;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: rgba(120,121,125,0.7);
  font-size: 0.875rem;
  line-height: 1.25;
  margin-left: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #FFFFFF;
  border: 1px solid #989A9C;
  border-radius: 50%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
  color: #081D17;
  border-color: #E0E1E2;
  color: #989A9C;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={true}
    className="emotion-0 emotion-1"
    disabled={true}
    id="example"
    name="example"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <div
    checked={true}
    className="emotion-4 emotion-5"
    disabled={true}
  >
    <i
      className="ar ar-radio-checked emotion-2 emotion-3"
      fontSize="16px"
    />
  </div>
  <span
    className="emotion-6 emotion-7"
    color="text.disabled"
    fontSize="size4"
  >
    My RadioButton
  </span>
</label>
`;

exports[`<RadioButton /> properly renders a radio button 1`] = `
.emotion-6 {
  font-family: Motiva Sans;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #FFFFFF;
  border: 1px solid #989A9C;
  border-radius: 50%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  line-height: 1.25;
  margin-left: 0.5rem;
}

<label
  className="emotion-6 emotion-7"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={false}
  />
  <span
    className="emotion-4 emotion-5"
    color="text.default"
    fontSize="size4"
  >
    My RadioButton
  </span>
</label>
`;

exports[`<RadioButton /> properly renders a radio button with an icon 1`] = `
.emotion-8 {
  font-family: Motiva Sans;
  margin-top: 1rem;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 1.25;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  opacity: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #FFFFFF;
  border: 1px solid #989A9C;
  border-radius: 50%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 1rem;
  width: 1rem;
}

.emotion-4 {
  color: #989A9C;
  font-size: 24px;
  margin-left: 0.5rem;
}

.emotion-6 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  line-height: 1.25;
  margin-left: 0.25rem;
}

<label
  className="emotion-8 emotion-9"
>
  <input
    aria-invalid={false}
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    id="example"
    name="example"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <div
    checked={false}
    className="emotion-2 emotion-3"
    disabled={false}
  />
  <i
    className="ar ar-cr-logo emotion-4 emotion-5"
    color="icon.default"
    fontSize="24px"
  />
  <span
    className="emotion-6 emotion-7"
    color="text.default"
    fontSize="size4"
  >
    My RadioButton
  </span>
</label>
`;
